src/components/Questionnaire/QuestionnaireForm.tsx

```
@@ -541,6 +570,438 @@ export function QuestionnaireForm({
+
+    const matchingRequests = originalPayload.requests.filter(
+      (req) => req.reference_id === "diagnosis",
+    );
+    if (matchingRequests.length === 0) return;
+
+    // Extract new datapoints & remove offline-ids
+    const newDatapoints: any[] = [];
+    matchingRequests.forEach((req) => {
+      req.body?.datapoints?.forEach((dp: any) => {
+        const copy = { ...dp };
+        if (copy.id?.startsWith("offline-")) delete copy.id;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)